File: Assignment41_5.c
#include <stdio.h>

///////////////////////////////////////////////////////////////////////////////
// Function Name : Mult
// Description   : Return product of digits using recursion
// Input         : Integer
// Output        : Product of digits
// Author        : Raj Samir Jadhav
// Date          : 06/12/2025
//////////////////////////////////////////////////////////////////////////////

int Mult(int iNo)
{
    static int iProd = 1;             // Store result

    if(iNo != 0)                     // Base condition
    {
        iProd = iProd * (iNo % 10);  // Extract digit and multiply
        iNo = iNo / 10;
        Mult(iNo);                  // Recursive call
    }

    return iProd;
}

int main()
{
    int iValue = 0, iRet = 0;

    printf("Enter number : ");
    scanf("%d",&iValue);

    iRet = Mult(iValue);

    printf("%d\n", iRet);

    return 0;
}

File: Assignment41_5.java
import java.util.Scanner;

///////////////////////////////////////////////////////////////////////////////
// Function Name : Mult
// Description   : Return product of digits using recursion
// Input         : Integer
// Output        : Product of digits
// Author        : Raj Samir Jadhav
// Date          : 06/12/2025
///////////////////////////////////////////////////////////////////////////////
class Pattern
{
    int Mult(int iNo)
    {
        if(iNo == 0)
        {
            return 1;                            // Base result for recursion
        }
        else
        {
            return (iNo % 10) * Mult(iNo / 10);  // Recursive call
        }
    }
}

class Assignment41_5
{
    public static void main(String[] args)
    {
        Scanner sobj = new Scanner(System.in);

        System.out.print("Enter number : ");
        int value = sobj.nextInt();

        Pattern pobj = new Pattern();
        int result = pobj.Mult(value);

        System.out.println(result);

        sobj.close();
    }
}
